### ISSUE TITLE
flameshot window does not correctly return focus

### ISSUE THREAD

**Issue by ndwarshuis** [xmonad/xmonad#218]
### Problem Description

This problem only seems to happen when `focusFollowsMouse = False`.

I am using a program called flameshot to capture my screen. When I run `flameshot gui` (which captures a specific area) a semi-transparent fullscreen window overlays an entire workspace and allows the capture area to be selected. When this window disappears, I expect the focus to return to the last window (independent of mouse position). What instead happens depends on where the mouse is relative to the last focused window:
1. If the mouse is over the last focused window, behavior is almost normal; the window has red borders (if using default config) and the keyboard appears to input correctly as if the window has focus. The only abnormal behavior is that urxvt will be dim (my urxvt is configured to gray-out when it loses focus).
2. If the mouse is over another window, the last focused window appears to retain focus (as in its borders are red if using a default config) but the keyboard input goes to the window under the mouse. If I move the mouse to a new window the keyboard input goes to that new window (as if `focusFollowsMouse = True`). Clicking on a window or using the keyboard to move focus (eg with `focusUp` or `focusDown` functions in XMonad) resumes normal behavior. Steps to reproduce:
   1. open two terminal windows
   2. move the mouse over the unfocused window
   3. run `flameshot gui` (from the prompt or terminal, doesn't matter)
   4. press escape to close flameshot
   5. type something (it should appear in the terminal under the mouse)
   6. move the cursor over the focused window
   7. type something (it should appear in the window now under the mouse)

I'm guessing the fix will involve special handling of a window property, but I don't know them well enough to propose anything. Here are the window properties of the flameshot window:

```
_NET_WM_USER_TIME(CARDINAL) = 26258262
_NET_WM_STATE(ATOM) = _NET_WM_STATE_ABOVE, _NET_WM_STATE_FULLSCREEN, _NET_WM_STATE_STAYS_ON_TOP
WM_TRANSIENT_FOR(WINDOW): window id # 0x80000a
_NET_WM_ICON_NAME(UTF8_STRING) = 
XdndAware(ATOM) = BITMAP
WM_NAME(STRING) = 
_NET_WM_NAME(UTF8_STRING) = "flameshot"
_MOTIF_WM_HINTS(_MOTIF_WM_HINTS) = 0x2, 0x1, 0x0, 0x0, 0x0
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_UTILITY, _KDE_NET_WM_WINDOW_TYPE_OVERRIDE, _NET_WM_WINDOW_TYPE_NORMAL
_XEMBED_INFO(_XEMBED_INFO) = 0x0, 0x1
WM_CLIENT_LEADER(WINDOW): window id # 0x80000a
WM_HINTS(WM_HINTS):
		Client accepts input or input focus: True
		Initial state is Normal State.
		window id # of group leader: 0x80000a
WM_CLIENT_MACHINE(STRING) = "petrucci4prez"
_NET_WM_PID(CARDINAL) = 139643
_NET_WM_SYNC_REQUEST_COUNTER(CARDINAL) = 8388638
WM_CLASS(STRING) = "flameshot", "flameshot"
WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW, WM_TAKE_FOCUS, _NET_WM_PING, _NET_WM_SYNC_REQUEST
WM_NORMAL_HINTS(WM_SIZE_HINTS):
		user specified size: 4480 by 1440
		window gravity: Static
```
Flameshot version
```
Flameshot v0.6.0
Compiled with Qt 5.12.1
```

There was also a similar issue in i3 (see [this issue](https://github.com/lupoDharkael/flameshot/issues/58) and [this commit](https://github.com/i3/i3/commit/a06e66982bc921d05b68b75f8e12323c284233d0))

### Configuration File

Please include the smallest configuration file that reproduces the
problem you are experiencing:

```haskell
module Main (main) where

import           System.Exit
import           XMonad
import           XMonad.Prompt
import           XMonad.Prompt.ConfirmPrompt
import           XMonad.Prompt.Shell
import           XMonad.Util.EZConfig

main =
  xmonad $ def
    { modMask    = mod4Mask
    , focusFollowsMouse = False
    }

    `additionalKeysP`
      [ ("M-S-q",   confirmPrompt myXPConfig "exit" (io exitSuccess))
      , ("M-p",     shellPrompt myXPConfig)
      ]

myXPConfig = def
  { position          = Top
  , alwaysHighlight   = True
  , promptBorderWidth = 0
  , font              = "xft:monospace:size=9"
  }

```

### Checklist

  - [x] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [x] I tested my configuration with [xmonad-testing](https://github.com/xmonad/xmonad-testing)

**Comment by karthikiyengar** [comment]
@ndwarshuis - Also, I was wondering if keyboard shortcuts are functioning correctly with flameshot under xmonad. They don't work at all for me.

**Comment by ndwarshuis** [comment]
@karthikiyengar those work for me, do you have the daemon running? My xinitrc has the line `flameshot &` which is required to make `flameshot gui` work at all. There should also be a dbus interface (org.dharkeal.Flameshot).

**Comment by karthikiyengar** [comment]
@ndwarshuis - Yes, but no dice. I launch xmonad using a desktop entry 
```
[Desktop Entry]
Name=XMonad
Encoding=UTF-8
Comment=Lightweight tiling window manager
Exec=/usr/local/bin/xmonad.start
Type=XSession
```
Where I've tried both `exec dbus-launch xmonad` and `xmonad` as the startup entry. There does seem to be a relevant dbus interface, but keyboard actions don't do anything. 

Interestingly, the AppImage version seems to work okay, although icons are missing. I've created a corresponding issue in the flameshot repo: https://github.com/lupoDharkael/flameshot/issues/773
